pkgs/development/libraries/openmpi/default.nix

```diff
@@ -40,21 +52,28 @@ in stdenv.mkDerivation rec {
     find -name "Makefile.in" -exec sed -i "s/\`date\`/$ts/" \{} \;
   '';
 
-  buildInputs = with stdenv; [ gfortran zlib ]
+  buildInputs = with stdenv; [ gfortran zlib libfabric pmix ucx ]
     ++ lib.optionals isLinux [ libnl numactl ]
     ++ lib.optionals cudaSupport [ cudatoolkit ]
     ++ [ libevent hwloc ]
     ++ lib.optional (isLinux || isFreeBSD) rdma-core;
 
   nativeBuildInputs = [ perl ];
 
-  configureFlags = with stdenv; lib.optional (!cudaSupport) "--disable-mca-dso"
+  configureFlags = with stdenv; [
+    "--with-cma"
+    "--with-hwloc=${hwloc.dev}"
+    "--with-ofi=${libfabric}"
+    "--with-pmi=${pmix}"
```

Comment:
I have tried to build with pmix in the past but could not get it work properly with slurm. Did you try if you can run openmpi binaries with `srun` instead of `mpirun`?
In general I would say pmix is not strictly necessary as openmpi already works nicely with slurm and detects the resource allocations properly. From how i understand pmix it is designed for large mpi jobs (>100 nodes).

Reply:
you're right, i've thought it worked, but i probably mistakenly tested mpirun multiple times?! will look into this.